core/market/src/db/dao/agreement.rs

```diff
@@ -126,31 +172,59 @@ impl<'c> AgreementDao<'c> {
                 })?
             }
 
-            diesel::update(dsl::market_agreement.filter(dsl::id.eq(&id)))
-                .set(dsl::state.eq(AgreementState::Approved))
-                .execute(conn)?;
+            update_state(conn, &id, &AgreementState::Approved)
+                .map_err(|e| StateError::DbError(e.into()))?;
```

Comment:
important: we also want to set `approved_date` here! (or `approved_ts` if you decide to refactor it)

Reply:
This is bigger change, than you might think. We need `approved_date` to match on both Provider and Requestor side, because we want to sign them together with whole Agreement. So changes in negotiation protocol are needed.
As long as we have `None` here and we don't lie to users, this can wait.
Issue for this:  #814